packages/prerelease-registry/functions/_middleware.ts

```diff
@@ -0,0 +1,9 @@
+export const onRequestOptions: PagesFunction = async () => {
+  return new Response(null, {
+    headers: {
+      "Access-Control-Allow-Origin": "*",
+      "Access-Control-Allow-Methods": "GET,OPTIONS",
+      "Access-Control-Max-Age": "86400",
+    },
+  });
+};
```

Comment:
This is another way of doing it that should mean you don't need the change in `getArtifactForWorkflowRun`.

```suggestion
export const onRequest: PagesFunction = async ({ next }) => {
  const response = await next()
  response.headers.set("Access-Control-Allow-Origin", "*");
  response.headers.set("Access-Control-Allow-Methods", "GET,OPTIONS");
  response.headers.set("Access-Control-Max-Age", "86400");
  return response;
};
```

Reply:
Maybe, IDK. Maybe an `OPTIONS` call would be weird. If that doesn't work, you could have an `onRequestOptions` and `onRequest` in this file: the first returns what you had. The second returns what I suggested. Then that'll cover all endpoints.

Reply:
oh nice! yeah, I think we need to keep options returning the response, and then just adding headers to everything else